src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -278,14 +278,15 @@ async def sink_buffer(sample: Sample[Quantity]) -> None:
             asyncio.create_task(self._resampler.resample(), name="resample")
         )
 
-    def __len__(self) -> int:
+    @property
+    def valid_size(self) -> int:
```

Comment:
I find `valid_size`  a bit unclear, for me it sounds it's talking about the validity of the **size**, not the entries. Also usually size if related to the amount of memory an object takes, this should still be a length, not a size., I'm not sure if it is too much, but maybe `len_of_valid_entries` or if we want a short version `len_valid`.

I'm even wondering if we need this method in the buffer/window. If this is the length of some slice or something you can get from the window, can't we implement the `len()` in that object instead?

Like if you can do `x = window.get_something()` and `len(x) == window.len_valid()`, when we don't really need `len_valid()` at all. Maybe we can have a quick chat about this.

Reply:
Good point, `len_valid` sounds like a better name. 

> Like if you can do x = window.get_something() and len(x) == window.len_valid(), when we don't really need len_valid() at all. Maybe we can have a quick chat about this.

If `get_something` is the new `window` function, then `len(x)` would only be equal to `len_valid` if there are no gaps, otherwise former would be larger as the window returns all values including gap-like entries while latter only counts the number of valid samples. I think both kind of lengths have their own valid use-cases and I am actually considering adding the other length to (e.g. `len_covered`). 

Completely removing it would at the moment not be trivial since it's used in several places already (e.g. moving window, periodic feature extractor).